llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -139,6 +140,10 @@ static cl::opt<bool>
     DisableDebugInfoPrinting("disable-debug-info-print", cl::Hidden,
                              cl::desc("Disable debug info printing"));
 
+static cl::opt<bool>
+    EmbedBitcodeFinal("embed-bitcode-final", cl::NotHidden,
+                             cl::desc("Embed post-optimisation bitcode"));
```

Comment:
"post-optimisation" could do with being clearer. As you mentioned, there are two waves of optimisations: platform agnostic and platform specific.

How about:
> Embed the final bitcode after all optimisations have run (including platform-specific backend passes).

Reply:
I think "optimisations" is actually too specific, since there can be non-optimising transformations (e.g. canonicalisations). Also, should we not say "Embed final IR as bitcode" rather than "embed bitcode"?

Reply:
Fixed in b7e0f6e